File: competence-database/src/main/scala/uzuzjmd/competence/owl/dao/CourseContext.scala
package uzuzjmd.competence.owl.dao

import uzuzjmd.competence.owl.access.{CompOntologyAccessJenaImpl, CompOntologyManager, CompOntologyAccessScala}
import uzuzjmd.competence.owl.ontology.CompOntClass
import uzuzjmd.competence.owl.ontology.CompObjectProperties

case class CourseContext(comp: CompOntologyManager, val name: String) extends CompetenceOntologyDao(comp, CompOntClass.CourseContext, name) {

  @Override
  protected def deleteMore() {
    //TODO
  }

  @Override
  protected def persistMore() {

  }

  def getFullDao(): CompetenceOntologyDao = {
    return this
  }

  def getLinkedCompetences(): List[Competence] = {
    if (name.equals("0") || name.equals("n0")) {
      return List.empty
    }
    return getAssociatedSingletonDaosAsRange(CompObjectProperties.CourseContextOf, classOf[Competence])
  }

  def getLinkedUser(): List[User] = {
    if (name.equals("0") || name.equals("n0")) {
      return List.empty
    }
    return getAssociatedStandardDaosAsDomain(CompObjectProperties.belongsToCourseContext, classOf[User])
  }

  def getMoodleId(): java.lang.Integer = {
    return Integer.parseInt(name.replace("n", ""))
  }

  override def equals(courseContext: Any): Boolean = {
    return courseContext.asInstanceOf[CourseContext].getId.equals(getId)
  }

}

File: competence-database/src/main/scala/uzuzjmd/competence/monopersistence/daos/CourseContext.java
package uzuzjmd.competence.monopersistence.daos;

import uzuzjmd.competence.persistence.ontology.CompObjectProperties;
import uzuzjmd.competence.persistence.ontology.Contexts;

import java.util.List;

/**
 * Created by dehne on 11.01.2016.
 */
public class CourseContext extends DaoAbstractImpl {

    public String requirement;

    public CourseContext(String id) {
        super(id);
        this. requirement = "";
    }

    public CourseContext(String id, String requirement) {
        super(id);
        this.requirement = requirement;
    }

    public CourseContext(Contexts context) {
        super(context.name());
    }

    public List<Competence> getLinkedCompetences() throws Exception {
        return getAssociatedDaosAsDomain(CompObjectProperties.CourseContextOf, Competence.class);
    }


    public List<User> getLinkedUser() throws Exception {
        return getAssociatedDaosAsDomain(CompObjectProperties.belongsToCourseContext, User.class);
    }

    public String getRequirement() {
        return requirement;
    }

    public void setRequirement(String requirement) {
        this.requirement = requirement;
    }

    @Override
    public CourseContext getFullDao() throws Exception {
        return super.getFullDao();
    }
}
